### PR TITLE
feat: add an option to enable GitHub discussion for a release

### PR DESCRIPTION
**Pull request by LinuxSuRen**
I'm not sure if this is appropriate to add a GitHub-specific option into the SCM struct. Please let me know if you have a better idea.

### CODE REVIEW

File: scm/driver/github/release.go [hunk 1]
```diff
@@ -26,12 +26,14 @@ type release struct {
...
-	Commitish   string `json:"target_commitish"`
-	Draft       bool   `json:"draft"`
-	Prerelease  bool   `json:"prerelease"`
+	Title                  string `json:"name"`
+	Description            string `json:"body"`
+	Tag                    string `json:"tag_name"`
+	Commitish              string `json:"target_commitish"`
+	Draft                  bool   `json:"draft"`
+	Prerelease             bool   `json:"prerelease"`
+	DiscussionCategoryName string `json:"discussion_category_name"`
```
Comment (ankitm123) [inline scm/driver/github/release.go:35]: Probably also want to add omitempty, if it's only a github specific feature?

Comment (LinuxSuRen) [reply scm/driver/github/release.go:35]: The discussion feature is not enabled by default.

File: scm/driver/github/testdata/release.json [hunk 2]
```diff
@@ -15,6 +15,7 @@
   "prerelease": false,
```
Comment (ankitm123) [inline scm/driver/github/testdata/release.json:15]: golden files?

**Comment by msvticket** [comment]
Seems abandoned by the developer

**Pull request closed without merging**